### PR TITLE
Support Ordering Dependent Duplicate storage volumes

### PR DESCRIPTION
**Pull request by kremlin-**
supports order dep. dupe. volumes for file/block

### CODE REVIEW

File: SoftLayer/CLI/block/duplicate.py [hunk 1]
```diff
@@ -54,9 +54,15 @@
               type=click.Choice(['hourly', 'monthly']),
               default='monthly',
               help="Optional parameter for Billing rate (default to monthly)")
+@click.option('--dependent-duplicate',
+              type=click.BOOL,
+              default=False,
+              help='Whether or not this duplicate will be a dependent duplicate '
+                    'of the origin volume (default to false)')
```
Comment (allmightyspiff) [inline SoftLayer/CLI/block/duplicate.py]: `show_default=True` option will print out the default in the help message, so you don't have to specify here.

https://click.palletsprojects.com/en/7.x/options/

File: SoftLayer/CLI/file/duplicate.py [hunk 2]
```diff
@@ -52,9 +52,16 @@
               type=click.Choice(['hourly', 'monthly']),
               default='monthly',
               help="Optional parameter for Billing rate (default to monthly)")
+@click.option('--dependent-duplicate',
+              type=click.BOOL,
+              default=False,
+              help='Whether or not this duplicate will be a dependent duplicate'
+                    'of the origin volume (default to false)')
```
Comment (allmightyspiff) [inline SoftLayer/CLI/file/duplicate.py]: Use `show_default=True`

File: SoftLayer/managers/block.py [hunk 3]
```diff
@@ -348,6 +351,9 @@ def order_duplicate_volume(self, origin_volume_id, origin_snapshot_id=None,
         if origin_snapshot_id is not None:
             order['duplicateOriginSnapshotId'] = origin_snapshot_id
 
+        if dependent_duplicate:
+            order['isDependentDuplicateFlag'] = 1
```
Comment (allmightyspiff) [inline SoftLayer/managers/block.py]: Would be simpler to just have `order['isDependentDuplicateFlag'] = dependent_duplicate` and remove the if statement

File: SoftLayer/managers/file.py [hunk 4]
```diff
@@ -289,6 +292,9 @@ def order_duplicate_volume(self, origin_volume_id, origin_snapshot_id=None,
         if origin_snapshot_id is not None:
             order['duplicateOriginSnapshotId'] = origin_snapshot_id
 
+        if dependent_duplicate:
+            order['isDependentDuplicateFlag'] = 1
```
Comment (allmightyspiff) [inline SoftLayer/managers/file.py]: Would be simpler to just have order['isDependentDuplicateFlag'] = dependent_duplicate and remove the if statement

File: SoftLayer/managers/storage_utils.py [hunk 5]
```diff
@@ -903,6 +906,9 @@ def prepare_duplicate_order_object(manager, origin_volume, iops, tier,
     if volume_is_performance:
         duplicate_order['iops'] = iops
 
+    if dependent_duplicate:
+        duplicate_order['isDependentDuplicateFlag'] = 1
```
Comment (allmightyspiff) [inline SoftLayer/managers/storage_utils.py]: Would be simpler to just have order['isDependentDuplicateFlag'] = dependent_duplicate and remove the if statement

File: SoftLayer/managers/block.py [hunk 6]
```diff
@@ -321,6 +322,8 @@ def order_duplicate_volume(self, origin_volume_id, origin_snapshot_id=None,
         :param duplicate_snapshot_size: Snapshot space size for the duplicate
         :param hourly_billing_flag: Billing type, monthly (False)
             or hourly (True), default to monthly.
+        :param dependent_duplicate: Duplicate type, normal (False) or dependent
```
Comment (allmightyspiff) [inline SoftLayer/managers/block.py]: you can use 120 character lines.

File: SoftLayer/CLI/block/duplicate.py (hunk 1 above)
Comment (kremlin-) [reply SoftLayer/CLI/block/duplicate.py]: as we discussed, this is not possible

File: SoftLayer/managers/block.py (hunk 6 above)
Comment (kremlin-) [reply SoftLayer/managers/block.py]: fixed in 7a651b0

File: SoftLayer/managers/storage_utils.py (hunk 5 above)
Comment (kremlin-) [reply SoftLayer/managers/storage_utils.py]: as we discussed, this isn't possible

File: SoftLayer/managers/file.py (hunk 4 above)
Comment (kremlin-) [reply SoftLayer/managers/file.py]: as we discussed, this isn't possible

File: SoftLayer/managers/block.py (hunk 3 above)
Comment (kremlin-) [reply SoftLayer/managers/block.py]: as we discussed, this isn't possible

File: SoftLayer/CLI/block/duplicate.py (hunk 1 above)
Comment (kremlin-) [reply SoftLayer/CLI/block/duplicate.py]: fixed in e424ce1

File: SoftLayer/CLI/file/duplicate.py (hunk 2 above)
Comment (kremlin-) [reply SoftLayer/CLI/file/duplicate.py]: fixed in e424ce1

**Comment by kremlin-** [comment]
> Use the `show_default=True` click CLI option instead of having the default be in the help message.
> 
> Instead of using the `if dependent_duplicate` check, just set the isDependentDuplicateFlag to whatever depend_duplicate is. Since it has a default value of 0/False, that should be fine.

this should all be fixed now

**Pull request merged**